lib/sparse/src/index/search_context.rs

```diff
@@ -598,4 +603,107 @@ mod tests {
             InvertedIndexMmap::convert_and_save(&inverted_index_ram, &tmp_dir_path).unwrap();
         _prune_test(&inverted_index_mmap);
     }
+
+    /// Generates a non empty sparse vector
+    pub fn random_sparse_vector<R: Rng + ?Sized>(rnd_gen: &mut R, max_size: usize) -> SparseVector {
+        let size = rnd_gen.gen_range(1..max_size);
+        let mut tuples: Vec<(i32, f64)> = vec![];
+
+        for i in 1..=size {
+            let no_skip = rnd_gen.gen_bool(0.5);
+            if no_skip {
+                tuples.push((i as i32, rnd_gen.gen_range(0.0..100.0)));
+            }
+        }
+
+        // make sure we have at least one vector
+        if tuples.is_empty() {
+            tuples.push((
+                rnd_gen.gen_range(1..max_size) as i32,
+                rnd_gen.gen_range(0.0..100.0),
+            ));
+        }
+
+        SparseVector::from(tuples)
+    }
+
+    /// Generates a random inverted index with `num_vectors` vectors
+    fn random_inverted_index<R: Rng + ?Sized>(
+        rnd_gen: &mut R,
+        num_vectors: u32,
+        max_sparse_dimension: usize,
+    ) -> InvertedIndexRam {
+        let mut inverted_index_ram = InvertedIndexRam::empty();
+
+        for i in 1..=num_vectors {
+            let vector = random_sparse_vector(rnd_gen, max_sparse_dimension);
+            inverted_index_ram.upsert(i, vector);
+        }
+        inverted_index_ram
+    }
+
+    #[test]
+    fn next_min_partial_scan_test() {
+        let num_vectors = 100;
+        let max_sparse_dimension = 25;
+        let mut rnd = StdRng::seed_from_u64(42);
+        let is_stopped = AtomicBool::new(false);
+        let inverted_index_ram = random_inverted_index(&mut rnd, num_vectors, max_sparse_dimension);
+        let mut search_context = SearchContext::new(
+            SparseVector {
+                indices: vec![1, 2, 3],
+                values: vec![1.0, 1.0, 1.0],
+            },
+            3,
+            &inverted_index_ram,
+            &is_stopped,
+        );
+
+        let mut all_next_min_observed = vec![];
```

Comment:
I propose to use `HashSet` instead of `Vec`. With set it's guaranteed that all ids are unique and check for `len()` means iterating over all points without duplications